experimental/perf/scripts/setup_helper.py

```diff
@@ -0,0 +1,49 @@
+import random
+import string
+import subprocess
+
+
+def generate_random_dict(num_fields: int, field_size: int) -> dict:
+    """Generates a JSON-like dict with the specified number of fields and field sizes.
+
+    Args:
+        num_fields (int): The number of key-value pairs (fields) in the JSON.
+        field_size (int): The size (in characters) of the field values.
+
+    Returns:
+        str: A JSON-like string with the specified structure.
+    """
+
+    def random_key():
+        # Generate a random key with a length "field_size"  characters
+        return "".join(
+            random.choices(string.ascii_letters + string.digits + "_", k=field_size)
+        )
+
+    # Generate the specified number of fields
+    return {random_key(): random.randint(1, 10**6) for _ in range(num_fields)}
+
+
+def capture_sar_metrics(log_dir: str, iteration: int = 8):
+    """Captures sar system metrics in the background and saves them to log files.
+
+    Args:
+        log_dir (str): Directory where the log files will be saved.
+        iteration (int): Number of seconds to capture metrics. Default is 8.
+    """
+    commands = {
+        "cpu.log": ["sar", "-u", "ALL", "1", str(iteration)],
+        "mem.log": ["sar", "-r", "1", str(iteration)],
+        "network.sock.log": ["sar", "-n", "SOCK", "1", str(iteration)],
+        "network.dev.log": ["sar", "-n", "DEV", "1", str(iteration)],
+        "paging.log": ["sar", "-B", "1", str(iteration)],
+        "disk.log": ["sar", "-d", "-p", "1", str(iteration)],
+    }
+
+    processes = []
+    for log_file, command in commands.items():
+        log_path = f"{log_dir}/{log_file}"
+        process = subprocess.Popen(
+            command, stdout=open(log_path, "w"), stderr=subprocess.PIPE
+        )
+        processes.append(process)
```

Comment:
1. could you please use context manager for the files:`with open(...) as f:`currently you don't ever close the files.
2. could you use os.path.join or pathlib to create paths to be system agnostic?
3. would be good to add error handling 
4. do you need to manage the sub process left after the parent process is gone?

Reply:
about 2), I've updated with pathlib.
about 3), added try/catch.

about 1) and 4), the load test is designed in a way such that

1. it spawns the sar subprocess in the background for X seconds (default 8s) to capture system metrics
2. runs the load test on main thread, which sometimes run for 20s, sometimes for 5s, it depends.
3. summarizes with whatever metrics we gathered so far

Because the load test running on the main python process *may* exit before the sar sub processes, using context mgr isn't fitting.  Since these sub processes will exit on their own regardless of the parent process status (i tested it, no resource leak or orphaned processes), it is safe to spawn them in a "fire-and-forget" style.

Reply:
these are valid concerns, i added more comments and notes in the codes to elaborate this "fire and forget" approach